packages/components/src/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,181 @@
+      setText("");
+      setOptions(initialOptions);
+    }
+  }, [value]);
+
+  return (
+    <div className={styles.autocomplete}>
+      <InputText
+        value={text}
+        onChange={textChange}
+        placeholder={placeholder}
+        onFocus={handleTextFocus}
+        onBlur={handleTextBlur}
+        ref={inputRef}
+      />
+      <Menu
+        visible={optionsMenuVisible}
+        options={options}
+        activeOption={value}
+        onOptionSelect={selectOption}
+      />
+    </div>
+  );
+
+  function selectOption(chosenOption: Option) {
+    onSelectOption(chosenOption);
+    textChange(chosenOption.label);
+    setOptionsMenuVisible(false);
+  }
+
+  async function textChange(newValue: string) {
+    setOptionsMenuVisible(true);
+    setText(newValue);
+    const results = await getOptions(newValue);
+    setOptions(results);
+  }
+
+  function handleTextBlur() {
+    setOptionsMenuVisible(false);
+    if (value == undefined || value.label !== text) {
+      onSelectOption(undefined);
+    }
+  }
+
+  function handleTextFocus() {
+    setOptionsMenuVisible(true);
+  }
+}
+
+interface MenuProps {
+  visible: boolean;
+  options: Option[];
+  activeOption?: Option;
+  onOptionSelect(chosenOption: Option): void;
+}
+
+function Menu({ visible, options, activeOption, onOptionSelect }: MenuProps) {
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+  const optionMenuClass = classnames(styles.options, {
+    [styles.visible]: visible,
+  });
+
+  setupKeyListeners();
+
+  useEffect(() => setSelectedIndex(-1), [options]);
+
+  return (
+    <div className={optionMenuClass}>
+      {options.map((option, index) => {
+        const optionClass = classnames(styles.option, {
+          [styles.active]: index === selectedIndex,
+        });
+        return (
+          <button
+            className={optionClass}
+            key={option.value}
+            onMouseDown={handleMouseDown(option)}
+          >
+            <Text>
+              {activeOption && option.value === activeOption.value && (
+                <>
+                  <Icon name="checkmark" size="small" />{" "}
+                </>
+              )}
+              {option.label}
+            </Text>
+          </button>
+        );
+      })}
+    </div>
+  );
+
+  function handleMouseDown(option: Option) {
+    return () => {
+      onOptionSelect(option);
+    };
+  }
+
+  function setupKeyListeners() {
+    useEffect(() => {
+      const handleKeyDown = (event: { key: string }) => {
+        const optionsLength = options.length;
+        if (optionsLength === 0) {
+          return;
+        }
+        if (event.key === "ArrowDown" && selectedIndex < optionsLength - 1) {
+          setSelectedIndex(selectedIndex + IndexChange.Next);
+        }
+        if (event.key === "ArrowUp" && selectedIndex > -1) {
+          setSelectedIndex(selectedIndex + IndexChange.Previous);
+        }
+      };
+      document.addEventListener("keydown", handleKeyDown);
```

Comment:
I don't totally understand what this is doing. Does this mean that regardless of what has the focus a key down will move the highlight in the autocomplete menu?